frontend/src/libs/components/app/app.tsx

```diff
@@ -8,43 +8,45 @@ import {
   useEffect,
   useLocation,
   useMemo,
-  useNavigate,
   useState,
 } from '~/libs/hooks/hooks.js';
 import { socket as socketService } from '~/libs/packages/socket/socket.js';
+import { selectUser } from '~/slices/auth/selectors.js';
 import { actions as userActions } from '~/slices/users/users.js';
 
 const App: React.FC = () => {
   const { pathname } = useLocation();
   const dispatch = useAppDispatch();
   const [isWebSocketsConnected, setIsWebSocketsConnected] = useState(false);
-  const navigate = useNavigate();
   const { users, dataStatus } = useAppSelector(({ users }) => ({
     users: users.users,
     dataStatus: users.dataStatus,
   }));
 
+  const user = useAppSelector(selectUser);
+  const hasUser = Boolean(user);
```

Comment:
Not sure if this is very important, but to avoid creating a variable on every render, you can pass `isAuth={!!user}`